index.d.ts

```diff
@@ -0,0 +1,28 @@
+export interface Crop {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+export interface CropResult {
+  topCrop: Crop;
+}
+export interface CropBoost {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+  weight: number;
+}
+export interface CropOptions {
+  minScale?: number;
+  width: number;
+  height: number;
+  boost?: CropBoost[];
+  ruleOfThirds?: boolean;
+  debug?: boolean;
+}
+export const smartcrop: {
+  crop(image: HTMLImageElement | HTMLCanvasElement | HTMLVideoElement, options: CropOptions): Promise<CropResult>;
```

Comment:
Looks good, I guess here image could be CanvasImageSource?

https://developer.mozilla.org/en-US/docs/Web/API/CanvasImageSource

Reply:
Yep, it can